### PR TITLE
doc/devguide: add page on Stream Frame support - v3

### PR DESCRIPTION
**Pull request by jufajardini**
This document offers a perspective more useful for developers. Once Frame support has been merged, we'll also add pages for rule writers and users (Frame logging).

Previous PR: #6704 

Describe changes:
- rebase & build on top of https://github.com/OISF/suricata/pull/6713 (to benefit from `literalinclude`)
- add short note on what Frame means in Suri context
- use `literalinclude` for code snippets
- reorganize contents for more fluidity
- add a bit more explanation on `StreamSlice`
- update rules, images and other places where Frame, PDU, Record & StreamSlice were still too blurred
- remove Scope section (table of contents works for that)

TODO add pages about:
- rule writing
 - frame logging

Link to redmine ticket: https://redmine.openinfosecfoundation.org/issues/4697

Compiled version can be seen at http://stream-frame-support-v3.surge.sh/extending/detect/app-layer-frames.html

### CODE REVIEW

File: doc/devguide/extending/detect/app-layer-frames.rst [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+A code example from the `SMB` decoder - code excerpt from `PR #6713 <https://github.com/OISF/suricata/pull/6713/commits/88c1fea77f55284c74443eaaa230e0c21956433f>`_:
+
+.. literalinclude:: ../../../../rust/src/smb/smb.rs
+    :caption: rust/src/smb/smb.rs
+    :language: rust
+    :start-at: pub fn parse_tcp_data_ts<'b>
+    :end-at: nbss_hdr.length as i64, SMBFrameType::NBSSData as u8);
+    :lines: 1-3, 56-64
+    :dedent: 4
```
Comment (jufajardini) [inline doc/devguide/extending/detect/app-layer-frames.rst:84]: Not sure if this is the best option, for this specific code-snippet. If we cut portions of the code, the snippet looks a bit weird. But if we don't, there are just too many lines...

File: doc/devguide/extending/detect/app-layer-frames.rst [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+================
+
+.. note:: Stream Frames support is a work in progress. This means that names and some implementation details might still change, before we reach a more stable version of the code. The overall idea behind it is matured, though, so this documentation is still valid. Current progress: `Feat/appl records <https://github.com/OISF/suricata/pull/6713>`_
+
+Suricata's architecture and current state of parsers mean that the network traffic stream isn't readily available to the
+detection engine. Except for the application layer parsers, the Suricata engine itself only has access to higher level
+abstractions - *State* and *Transactions* (much heavier objects, that store *Frame* information in the form of *Request-Response* pairs).
+
+.. note:: For Suricata, *Frame* is a generic enough term that it can represent any unit of network data we are interested in, which could be comprised of one or several packages of other, lower level protocol(s).
+
```
Comment (jufajardini) [inline doc/devguide/extending/detect/app-layer-frames.rst:24]: Not sure if `note` is good, or if should just keep as normal text. Also not sure if this definition is good enough.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 5334

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata/pull/6928

**Pull request closed without merging**